frontend/hooks/data-fetching/course.ts

```diff
@@ -311,12 +310,13 @@ export async function createQuestion(
     );
 
     if (!res.ok) {
-        const error = await res.json();
-        throw {
-            payload: JSON.stringify(payload),
-            error: JSON.stringify(error),
-        };
+        if (res.status !== 409 && res.status !== 429) {
+            const error = await res.json();
+            logException(Error(JSON.stringify(error)), JSON.stringify(payload));
+        }
+        return new Promise((resolve) => resolve(res.status));
```

Comment:
No need to manually construct a promise. You can just return res.status or undefined. Because the function is marked async, [the value will be wrapped in a promise anyway](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#:~:text=Async%20functions%20always%20return%20a%20promise.%20If%20the%20return%20value%20of%20an%20async%20function%20is%20not%20explicitly%20a%20promise%2C%20it%20will%20be%20implicitly%20wrapped%20in%20a%20promise.)